lib/commands/backup.js

```diff
@@ -0,0 +1,230 @@
+
+function absolutePath(inputPath) {
+    return path.isAbsolute(inputPath) ? inputPath : path.join(process.cwd(), inputPath);
+}
+
+class BackupCommand extends Command {
+    run(argv) {
+        const instance = this.system.getInstance();
+
+        return this.ui.listr([{
+            task: this.initialize,
+            title: 'Checking configuration'
+        }, {
+            title: 'Backing up extensions',
+            task: this.backupExtensions
+        }, {
+            title: 'Backing up database',
+            task: this.backupDatabase
+        }, {
+            title: 'Backing up config files',
+            task: this.backupConfig
+        }, {
+            title: 'Backing up content folder',
+            task: this.backupContent.bind(this)
+        }, {
+            title: 'Writing backup file',
+            task: this.writeZipFile
+        }],{
+            system: this.system,
+            argv: argv,
+            instance: instance
+        });
+    }
+
+    initialize(ctx) {
+        const datetime = (new Date()).toJSON().substring(0, 10);
+
+        ctx.database = {};
+        ctx.saveLocation = process.cwd();
+
+        if (ctx.argv.output) {
+            ctx.saveLocation = absolutePath(ctx.argv.output);
+        }
+
+        try {
+            fs.ensureDirSync(ctx.saveLocation);
+            // After we're sure the directory exists, check to make sure we can write to it
+            fs.accessSync(ctx.saveLocation, fs.W_OK);
+        } catch (error) {
+            ctx.instance.ui.log('Ghost doesn\'t have permission to write to the output folder', 'red');
+            return Promise.reject(new errors.SystemError(error));
+        }
+
+        const saveRoot = ctx.saveLocation
+        ctx.saveLocation = path.join(saveRoot, `${ctx.instance.name}.backup.${datetime}.zip`);
+
+        let offset = 1;
+
+        while (fs.existsSync(ctx.saveLocation)) {
+            ctx.saveLocation = path.join(saveRoot, `${ctx.instance.name}.backup.${datetime}-${offset}.zip`);
+            offset += 1;
+        }
+
+        // The zip library used (adm-zip) has a known issue where the zip is structurally sound,
+        // but Windows has issues reading / unzipping it. To counteract this, all files are read
+        // and added via the `addFile` method, where the file permissions can be set, seemingly
+        // resolving the issue. See https://github.com/cthackers/adm-zip/issues/171
+        ctx.zipFile = new Zip();
+
+        if (ctx.instance.running()) {
+            ctx.instance.ui.log('Ghost is currently running. Backing up might take longer and slow down your site', 'yellow');
+            ctx.instance.loadRunningEnvironment(true);
+        } else {
+            ctx.instance.checkEnvironment();
+        }
+
+        return Promise.resolve();
+    }
+
+    backupContent(ctx, dontcare, folder) {
+        folder = folder || './content';
+
+        // path MUST be relative
+        if (folder.indexOf('./') !== 0) {
```

Comment:
The content folder path in the target config file can be absolute or relative.
If you add `./` to an absolute path, it won't work?

Reply:
@kirrg001 it won't work I think because if the path isn't relative then it will get saved into the zipfile as an absolute path, which can cause issues iirc when unzipping the file.

Reply:
I completely forgot that the contentPath is configurable 😨 I fixed the issue, but fixing the issue screwed up the zip file. Since everyone seems to be in favor of using `tar.gz` over `zip`, I'm not going to try and fix the new issue.

That being said, I think the content, no matter where it is configured to be for the instance, should be backed up to `{archiveRoot}/content`. When speccing the restore command, we need to be mindful of this, specifically how to react if the content-path (or 1 level up) doesn't exist.